addons/mail/static/src/models/composer/composer.js

```diff
@@ -682,15 +682,18 @@ function factory(dependencies) {
          * @param {string} mentionKeyword
          */
         _updateSuggestedChannelCommands(mentionKeyword) {
-            this.update({
-                suggestedChannelCommands: [[
-                    'replace',
-                    this.env.messaging.commands.filter(
-                        command => command.name.includes(mentionKeyword)
-                    )
-                ]],
+            const commands = this.env.messaging.commands.filter(command => {
+                if (!command.name.includes(mentionKeyword)) {
+                    return false;
+                }
+                if (command.channel_types) {
+                    return command.channel_types.includes(this.thread.channel_type);
+                }
+                return true;
             });
 
+            this.update({ suggestedChannelCommands: [['replace', commands]] });
+
```

Comment:
No newlines (yes, even the one not in diff).
Newlines inside function should be used very rarely.

Reply:
Done